### DISCUSSION TITLE
Tables created in Tabular Editor 3 do not sync back to Model in Power BI

### DISCUSSION THREAD

**Discussion by BradCurtisBoxCut** [TabularEditor/TabularEditor3#643] [🙋♀️TE3 Q&A]
Hi,
I am encountering a strange problem when creating new tables using a script via Tabular Editor. In a simple Power BI model sourced from an excel sheet, I am using a csx script to essentially do the following:

    maxTable = Model.AddTable("MaxDate",true);
    maxTable.SetAnnotation("PBI_NavigationStepName","Navigation");
    maxTable.SetAnnotation("PBI_ResultType","Date");

    string maxDateExpression = "let\n    maxDate = let\n    maxDate =    List.Max(<tablename>[<columnname>])\nin\n    maxDate as date\nin\n    maxDate";
    var mExpression = maxTable.Partitions.FirstOrDefault();
    mExpression.Query = maxDateExpression;
    mExpression.Mode = ModeType.Import;  
    var maxDateColumn = maxTable.AddDataColumn("MaxDate","MaxDate",dataType: DataType.DateTime);

(Where <tablename> is the name of the table in my model).

When this script runs, it adds the new table correctly inside tabular editor, and everything looks fine. When I save changes back to the model, Power BI correctly notifies me that there are pending changes, but when I go to apply them, my changes are immediately removed.

To confirm that the results of my script were correct, I could save the results of tabular editor as a .bim file, and see my new table there. Furthermore, in Power BI, I can save my model as a pbit file, open up the archive it creates, and add the json for my new table (from the .bim file) into the pbit archive. When loading the pbit with the new table json saved, it correctly appears inside power BI. 

Am I missing something here as to why this happens? Or can you think of a reason why this might not work?

Thanks

**Comment by otykier** [comment]
Creating/editing tables with external tools, in a model hosted in Power BI Desktop, is unfortunately [not supported by Microsoft](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#data-modeling-operations).

**Reply by BradCurtisBoxCut** [reply]
Thanks Daniel, I had a feeling this might be the case
